BLING package update

mmazloff opened the pull request:
## What changes does this PR introduce?
(Bug fix, feature, docs update, ...)
BLING default package is unchanged. But more options added (Silica cycling, new chem, new MLD diagnostic), and overall structure significantly cleaned. 
BLINGv1 (Galbraith et al., 2010) and the new NBLING are cleanly separated. 
BLINGv1 is consistent with DIC package inputs. 
 
## What is the current behaviour? 
(You can also link to an open issue here)
BLING BGC package

## What is the new behaviour 
(if this is a feature change)?
BLINGv1 and NBLING packages are cleanly separated. 

## Does this PR introduce a breaking change? 
(What changes might users need to make in their application due to this PR?)


## Other information:
CHANGES:
1) Obsolete files removed: bling_production.F, bling_remineralization.F, bling_dvm.F
2) gchem_forcing_sep.F is revised to call BLING or NBLING
3) bling_carbon_chem: Replacing apressure with AtmosP, which is default=1 and set to apressure in bling_fields_load if using EXF
4) bling_carbonate_init: Use silica tracer value instead of a constant value if using SiBLING
5) bling_carbonate_sys: Use silica tracer value instead of a constant value if using SiBLING
6) bling_diagnostic_init: Adding diagnostic for MLD, Adding diagnostics for silica when using SiBLING
7) bling_light: Adding new option to calculate light penetration. Dealing with ice here instead of in bling_readparms
8) bling_min_val: Keeping track of corrections to tracer fields for closing budgets
9) bling_mixed_layer: Changing the way MLD is calculated. For old threshold criterion, #define BLING_USE_THRESHOLD_MLD
10) BLING_OPTIONS: New options for updated carbon chemistry (from J. Lauderdale), mixed layer and light calculation.  Removed option to use EXF atm pressure and atm pCO2; these are now always used if available through EXF. Adding option for SiBLING (silica tracer) 
11) Added bling_bio: Original 6-tracer bling model: Combines bling_production.F and bling_remineralization.F
12) Added bling_bio_nitrogen: Current bling model, with nitrogen cycling (default has 8 tracers, options for additional tracers for silica and phytoplankton). Combines bling_production.F and bling_remineralization.F
13) BLING_VARS: Adding variable MLmix_max (max depth over which light and biomass can be mixed; default = 200m)
14) bling_main: Adding nutrients in runoff
Different order of tracers if using original 6-tracer bling code (USE_BLING_v0)
Call either bling_bio.F or bling_bio_nitrogen.F, depending on which version of the model we’re using. 
Adding option for SiBLING (silica tracer) 
15) bling_fields_load: Wind, fice, atmosp, atm_pco2 are read either from file or EXF

## Suggested addition to `tag-index`
(To avoid unnecessary merge conflicts, please don't update `tag-index`. One of the admins will do that when merging your pull request.)

jm-c commented:
@mmazloff  I cleaned up few things there. I looked briefly to the code and have 2 comments:
1) the new CPP option "USE_BLING_V0" is not in BLING_OPTIONS.h ; should appear there (I guess
 as #undef) with 2 line of comment. Did not check if there is other new CPP options that are not 
 documented but if there are, they should be added to BLING_OPTIONS.h too.
2) we will need to update the reference output of experiment "global_oce_biogeo_bling" in i 
 verification_other repo. We could decide to submit a PR there once this one is merge ?
And just to open the discussion: Might be good, as some point, to change the order of S/R argument
in pkg/bling, from "myIter, myTime, myThid" to "myTime, myIter, myThid" like every where in the remaining MITgcm code (except in pkg/dic, that we could change later). And since you updated all
pkg/bling code in this PR, we might take this opportunity to make this switch ?

mmazloff commented:
@jm-c 
1) #undef USE_BLING_V0  is in the new checked in BLING_OPTIONS.h. It is on line 14
2) We will fix to the order myTime, myIter, myThid 
Thanks!

jm-c commented:
@mmazloff you are right, I missed this one.

mmazloff commented:
@jm-c
the ordering of the S/R arguments have been brought into consitency with the rest of the MITgcm. It is now "myTime, myIter, myThid"

jm-c commented:
@mmazloff I run testreport with debug options that check S/R arguments so I think it's good now.
Just a detail: since you did not provide a suggested "tag-index addition", may be I could let you do this
(i.e., add documentation in tag-index) when we are ready to merge. Let me know.

mmazloff commented:
@jm-c Great! I am ready to see this merged. I can ad documentation to tag-index at that time. How much detail do you recommend. Here are the major changes. Is putting all this too much detail for a tag-index?


1) bling_carbon_chem: Replacing apressure with AtmosP, which is default=1 and set to apressure in bling_fields_load if using EXF
2) bling_carbonate_init: Use silica tracer value instead of a constant value if using SiBLING
3) bling_carbonate_sys: Use silica tracer value instead of a constant value if using SiBLING
4) bling_diagnostic_init: Adding diagnostic for MLD. Adding diagnostics for silica when using SiBLING
5) bling_light: Adding new option to calculate light penetration. Dealing with ice here instead of in bling_readparms
6) bling_min_val: Keeping track of corrections to tracer fields for budgets
7) bling_mixed_layer: Changing the way MLD is calculated. For old threshold criterion, #define BLING_USE_THRESHOLD_MLD
8) BLING_OPTIONS:  New options for updated carbon chemistry (from J. Lauderdale), mixed layer and light calculation.  Removed option to use EXF atm pressure and atm pCO2; these are now always used if available through EXF. Adding option for SiBLING (silica tracer) 
9) bling_bio: Original 6-tracer bling model. Combines bling_production.F and bling_remineralization.F
10) bling_bio_nitrogen: Current bling model, with nitrogen cycling (default has 8 tracers, options for additional tracers for silica and phytoplankton). Combines bling_production.F and bling_remineralization.F
11) BLING_VARS: Adding variable MLmix_max (max depth over which light and biomass can be mixed; default = 200m)
12) bling_main: Adding nutrients in runoff. Different order of tracers if using original 6-tracer bling code (USE_BLING_v0). Call either bling_bio.F or bling_bio_nitrogen.F, depending on which version of the model we’re using.  Adding option for SiBLING (silica tracer) 
13) bling_fields_load: Wind, fice, atmosp, atm_pco2 are read either from file or EXF
14) gchem_forcing_sep.F: updated for compatibility
15) ordering of the S/R arguments have been brought into consitency with the rest of the MITgcm. Now "myTime, myIter, myThid"
